### PR TITLE
[MBL-1459] Tab bar badging for PPO

### PR DESCRIPTION
**Pull request by stevestreza-ksr**
# 📲 What

This PR connects the badge on the tab bar to the User property that says there is a pending PPO action.

# 🛠 How

The User object has a `ppoHasAction` property now. When this field is true and the PPO feature flag is enabled, we show a dot badge rather than a count by passing an empty string.

There was a bunch of existing badging support that already worked with a string. This PR uses that existing badging with the same signals, and extends it where the PPO feature flag is enabled.

# 👀 See

| Before 🐛 | After 🦋 |
| --- | --- |
|  |  |

# ✅ Acceptance criteria

Tests should pass.

### CODE REVIEW

**Comment by nativeksr** [comment]
### SwiftFormat found issues:

| File | Rules |
| ---- | ----- |
| Library/ViewModels/RootViewModel.swift:272:1 | warning: (trailingClosures) Use trailing closure syntax where applicable. |
| Library/ViewModels/RootViewModel.swift:274:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Library/ViewModels/RootViewModelTests.swift:617:1 | warning: (indent) Indent code in accordance with the scope level. |
| Library/ViewModels/RootViewModelTests.swift:618:1 | warning: (indent) Indent code in accordance with the scope level. |
| Library/ViewModels/RootViewModelTests.swift:619:1 | warning: (indent) Indent code in accordance with the scope level. |


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by jovaniks — APPROVED** [review]
LGTM

File: Library/ViewModels/RootViewModel.swift [hunk 1]
```diff
@@ -457,7 +470,12 @@ private func tabData(forUser user: User?) -> TabBarItemsData {
 extension TabBarItemsData: Equatable {}
 extension TabBarItem: Equatable {}
 
-private func activitiesBadgeValue(with value: Int?) -> String? {
+private func activitiesBadgeValue(with value: Int?, hasPPOAction: Bool) -> String? {
```
Comment (ifosli) [inline Library/ViewModels/RootViewModel.swift:473]: I like this interpretation, but my understanding was that if PPO is enabled, we never want to show a count. So ex if I have PPO enabled, no PPO actions, but I do have activity notifications, I was expecting the dot to show with no count. Would you mind checking with Alison what she intended before you submit this?

File: Library/ViewModels/RootViewModelTests.swift [hunk 2]
```diff
@@ -611,6 +611,50 @@ final class RootViewModelTests: TestCase {
...
+      RemoteConfigFeature.pledgedProjectsOverviewEnabled.rawValue: false
+    ]
+
+    self.setBadgeValueAtIndexValue.assertValues([])
+    self.setBadgeValueAtIndexIndex.assertValues([])
+
+    withEnvironment(currentUser: user, remoteConfigClient: remoteConfig) {
+      self.vm.inputs.viewDidLoad()
+
+      self.setBadgeValueAtIndexValue.assertValues([nil])
```
Comment (ifosli) [inline Library/ViewModels/RootViewModelTests.swift:654]: I'm very confused by this test - shouldn't the value here be `54`, not nil? Would you mind explaining why this is intended or looking into why the test isn't working?

**Review by ifosli — CHANGES_REQUESTED** [review]
Code looks good, I just have a couple concerns around intended behavior (of the dot and of that last test). Let me know if I'm wrong about changes being needed and I'd be happy to approve.

File: Library/ViewModels/RootViewModelTests.swift (hunk 2 above)
Comment (stevestreza-ksr) [reply Library/ViewModels/RootViewModelTests.swift]: Good call, tests updated to include testing for a badge.

File: Library/ViewModels/RootViewModel.swift (hunk 1 above)
Comment (stevestreza-ksr) [reply Library/ViewModels/RootViewModel.swift:473]: Alison says we should show the count in the case where PPO is enabled, there is no PPO action, and there is activity count.

**Comment by stevestreza-ksr** [comment]
Feel free to merge on my behalf once a second approval lands. 😄

**Review by ifosli — APPROVED** [review]
Nice!

**Pull request merged**